File: rust/455.assign-cookies.rs
/*
 * @lc app=leetcode id=455 lang=rust
 *
 * [455] Assign Cookies
 */

// @lc code=start
impl Solution {
    pub fn find_content_children(g: Vec<i32>, s: Vec<i32>) -> i32 {
        if s.len() == 0 {
            return 0;
        }

        let mut g = g;
        let mut s = s;
        g.sort_unstable();
        s.sort_unstable();
        let mut content = 0;
        let (mut i, mut j): (usize, usize) = (0, 0);
        while i < g.len() && j < s.len() {
            if s[j] >= g[i] {
                content += 1;
                i += 1; 
            }
            j += 1;
        }
        content
    }
}
// @lc code=end

File: javascript/455.assign-cookies.js
/*
 * @lc app=leetcode id=455 lang=javascript
 *
 * [455] Assign Cookies
 */

// @lc code=start
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function (g, s) {
  let contented = 0;
  if (s.length) {
    g.sort((a, b) => a - b);
    s.sort((a, b) => a - b);
    let i = 0,
      j = 0;
    while (i < g.length && j < s.length) {
      if (s[j++] >= g[i]) {
        contented++;
        i++;
      }
    }
  }
  return contented;
};
// @lc code=end
